### PR TITLE
Feature/jwt passport

### PR DESCRIPTION
**Pull request by shamoons**

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -21,7 +21,23 @@ array-flatten@1.1.1:
   version "1.1.1"
```
Comment (gjanakiram) [inline yarn.lock]: Are we going to commit yarn.lock file also ?

Comment (shamoons) [reply yarn.lock]: Yes - it's best practice to commit lock files

https://stackoverflow.com/questions/44552348/should-i-commit-yarn-lock-and-package-lock-json-files

File: test/auth.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+describe("Basic Authentication with JWT", () => {
+  it('Should login properly', function () {
+    return request(app)
+      .post('/login')
+      .send({
+        name: 'shamoon',
+        password: 'password'
+      })
+      .expect(200)
```
Comment (gopalakrishnanu) [inline test/auth.js]: Also check that the response has a jwt

Comment (gopalakrishnanu) [reply test/auth.js]: Also add a test cases to validate that un authenticated access to /secret will fail and authenticated access will pass

File: routes/index.js [hunk 3]
```diff
@@ -15,4 +15,12 @@ module.exports = (app) => {
     associations: true,
     pagination: false
   })
+
+  userResource.list.fetch.before((req, res, context) => {
+    passport.authenticate('jwt', { session: false })
+  })
+
+  app.get("/secret", passport.authenticate('jwt', { session: false }), (req, res) => {
```
Comment (gopalakrishnanu) [inline routes/index.js]: What does passport.authenticate() do?

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[*.js]
```
Comment (gopalakrishnanu) [inline .editorconfig]: Add to .gitignore

**Pull request merged**